libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,430 @@
+
+        // Assertions
+        assertEquals(1, evcsGetUserVCsDto.vcs().size());
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact invalidRetrieveVcRequestReturnsWith400(PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("VCS exist")
+                .given("EVC has users VC")
+                .uponReceiving("A request to get VCS")
+                .path("/vcs/" + INVALID_USER_ID)
+                .query("state=" + PENDING_RETURN)
+                .method("GET")
+                .headers(
+                        Map.of(
+                                "x-api-key",
+                                EVCS_API_KEY,
+                                "Authorization",
+                                "Bearer " + TEST_EVCS_ACCESS_TOKEN))
+                .willRespondWith()
+                .status(400)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "invalidRetrieveVcRequestReturnsWith400")
+    void testRetrieveVcRequestReturns400(MockServer mockServer) {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        assertThrows(
+                EvcsServiceException.class,
+                () -> {
+                    evcsClient.getUserVcs(
+                            INVALID_USER_ID, TEST_EVCS_ACCESS_TOKEN, VC_STATES_FOR_QUERY);
+                });
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact invalidRetrieveVcRequestReturnsWith404(PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("VCS exist")
+                .given("EVC has users VC")
+                .uponReceiving("A request to get VCS")
+                .path("/vcs/" + INVALID_USER_ID)
+                .query("state=" + PENDING_RETURN)
+                .method("GET")
+                .headers(
+                        Map.of(
+                                "x-api-key",
+                                EVCS_API_KEY,
+                                "Authorization",
+                                "Bearer " + TEST_EVCS_ACCESS_TOKEN))
+                .willRespondWith()
+                .status(404)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "invalidRetrieveVcRequestReturnsWith404")
+    void testRetrieveVcRequestReturns404(MockServer mockServer) throws Exception {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        var userVcs =
+                evcsClient.getUserVcs(INVALID_USER_ID, TEST_EVCS_ACCESS_TOKEN, VC_STATES_FOR_QUERY);
+        assertEquals(Collections.emptyList(), userVcs.vcs());
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact invalidRetrieveVcRequestReturnsWith401(PactDslWithProvider builder) {
```

Comment:
perhaps better to rephrase as `invalidApiKeyReturns401` to be really clear about what you're doing